drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/update_referral_posting_job.rb

```diff
@@ -24,21 +24,35 @@ module HmisExternalApis::AcHmis
   class UpdateReferralPostingJob < ApplicationJob
     include HmisExternalApis::AcHmis::ReferralJobMixin
 
-    # @param identifier [String]  HmisExternalApis::AcHmis::ReferralPosting.identifier
-    # @param status [Integer] new status
-    # @param referral_result [Integer] Only for denials. Value from HUD list 4.20.D.
-    # @param url [String]
-    def perform(identifier:, status:, url:, referral_result: nil)
-      raise 'Invalid status. Expected one of: [Closed(13), AcceptedPending(18), DeniedPending(19), Accepted(20), Denied(21)]' unless [13, 18, 19, 20, 21].include?(status)
+    VALID_STATUS_IDS = HmisExternalApis::AcHmis::ReferralPosting.statuses.values_at(:closed_status, :accepted_pending_status, :denied_pending_status, :accepted_status, :denied_status).to_set
+
+    # @param posting_id [Integer]  HmisExternalApis::AcHmis::ReferralPosting.identifier
+    # @param posting_status_id [Integer] new status
+    # @param requested_by [String]
+    # @param referral_result_id [Integer] Only for denials. Value from HUD list 4.20.D.
+    # @param denied_reason_id [Integer] required when Posting status is Denied Pending
+    # @param denied_reason_text [String]
+    # @param status_note [String]
+    # @param contact_date [String] required when Posting status is Denied Pending or Accepted Pending
+    def perform(posting_id:, posting_status_id:, requested_by:, denied_reason_id: nil, denied_reason_text: nil, status_note: nil, contact_date: nil, referral_result_id: nil)
+      raise 'Invalid status. Expected one of: [Closed(13), AcceptedPending(18), DeniedPending(19), Accepted(20), Denied(21)]' unless VALID_STATUS_IDS.include?(posting_status_id)
```

Comment:
Is it worth maybe:
```suggestion
      raise "Invalid status. Expected one of: #{VALID_STATUS_IDS.inspect}" unless VALID_STATUS_IDS.include?(posting_status_id)
```

I guess that doesn't include the descriptions, so maybe not so useful.
To prevent having to update this in the future?

Reply:
Good point, yeah I think we may want to revisit and enhance validation when we [hook this up to interface](https://www.pivotaltracker.com/story/show/185276764)